djconnectwise/sync.py

```diff
@@ -929,10 +935,15 @@ def _assign_field_data(self, instance, json_data):
             members = list(models.Member.objects.filter(
                 id__in=json_data.get('members')))
 
-        instance.save()
+        old_members = list(instance.members.all())
```

Comment:
This block does what it's supposed to do but I don't think it's ideal to keep it in the `assign_field_data` method. The purpose of this method is to set data on the model not to check if the stored data is different than the new data from the API. 

In my mind, the most appropriate place for checking if data has changed is the `_is_instance_changed`. 

If you give `TeamSynchronizer` its own `_is_instance_changed` method and pass in `api_instance` from `update_or_create_instance`, I think you should be easily able to perform the same check to determine if members have changed.